src/analysis.jl

```diff
@@ -93,10 +87,13 @@ end
 function globals(warn, call)
   c = code(call)[1]
   eachline(c) do line, ex
-    (isexpr(ex, :(=)) && isexpr(ex.args[2], GlobalRef)) || return
-    ref = ex.args[2]
-    isconst(ref.mod, ref.name) ||
+    ex isa Expr || return
```

Comment:
GlobalRefs can be top level as well, so this might be best expressed with `prewalk`.

Reply:
Not sure what you mean?

Reply:
As in, this will catch statements of the form `%5 = foo(Base.pi)` but not of the form `%4 = Base.pi; %5 = foo(%4)`

Reply:
Ah, that's because this pass actually runs on the *un*optimised code_typed, not the SSA IR of `@code_typed`.